src/ngraph/builder/quantization_util.hpp

```diff
@@ -99,6 +99,59 @@ namespace ngraph
                     (std::pow(2, -24) * static_cast<double>(max_abs32 / max_abs8)));
                 return scale;
             }
+
+            static inline float get_quantize_scale(const std::shared_ptr<Node> min_input,
+                                                   const std::shared_ptr<Node> max_input,
+                                                   bool is_signed)
+            {
+                auto min_input_const_op = std::static_pointer_cast<ngraph::op::Constant>(min_input);
+                auto max_input_const_op = std::static_pointer_cast<ngraph::op::Constant>(max_input);
+                float input_min_range =
+                    *(static_cast<float const*>(min_input_const_op->get_data_ptr()));
+                float input_max_range =
+                    *(static_cast<float const*>(max_input_const_op->get_data_ptr()));
+
+                // begin code copied and pasted from
+                // github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/quantize_op.cc
+                float min_range;
+                float max_range;
+                // If input_min_range and input_max_range are close,
+                // introduce a slightly larger delta between them.
+                min_range = std::min(0.0f, input_min_range);
+                const float epsilon =
+                    std::max(1.0f, std::max(fabsf(input_min_range), fabsf(input_max_range))) /
+                    100.0f;
+                max_range = std::max(input_max_range, min_range + epsilon);
+                max_range = std::max(0.0f, max_range);
+                // end code copied and pasted from
+                // github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/quantize_op.cc
+                const float max_abs = std::max(std::abs(min_range), std::abs(max_range));
+                const float target_range =
+                    static_cast<float>((is_signed ? std::pow(2, 7) : std::pow(2, 8)) - 1);
+                max_range = max_abs;
```

Comment:
this is hard coded to num_bits = 8, which works for now because we only support qi8 and qu8

We could have num_bits passed to this function, in case there are plans for qi16, qu16, q132 etc

Reply:
my intention is to add type traits (min, max) to each element type.  this way, you could have, for example, a int8 with range [-128, 127] and also qint8 (or some such name) with range [-127, 127].  so, why don't we pass `type` to this function and then call `is_signed` on that type?  you could also call `bitwidth` for now to do this calculation.